emuu-client/src/components/UserProfile/Profile.js

```diff
@@ -16,29 +18,89 @@ import {
+import Button from "@material-ui/core/Button";
+import Typography from "@material-ui/core/Typography";
+import { withStyles } from "@material-ui/core/styles";
+import { getOrientation } from "get-orientation/browser";
+import ImgDialog from "./imgDialog";
+import { getCroppedImg, getRotatedImage } from "./canvasUtils";
+import { styles } from "./styles";
+import HeaderPostLogin from "../NavbarPostLogin/HeaderPostLogin.js";
+import { uploadString } from "@firebase/storage";
+import { useHistory } from "react-router-dom";
+const ORIENTATION_TO_ANGLE = {
+  3: 180,
+  6: 90,
+  8: -90,
+};
+
+
+function Profile({ setVideo, video }, { classes }) {
   const [percent, setPercent] = useState(0);
   const displayName = localStorage.getItem("displayName");
   const docRef = doc(db, "Users", displayName);
+  const [imageSrc, setImageSrc] = React.useState(null);
+  const [croppedImageSrc, setCroppedImageSrc] = React.useState(null);
+  const [crop, setCrop] = useState({ x: 0, y: 0 });
+  const [rotation, setRotation] = useState(0);
+  const [zoom, setZoom] = useState(1);
+  const [croppedAreaPixels, setCroppedAreaPixels] = useState(null);
+  const [croppedImage, setCroppedImage] = useState(null);
+  const history = useHistory();
+
+  const onCropComplete = useCallback((croppedArea, croppedAreaPixels) => {
+    setCroppedAreaPixels(croppedAreaPixels);
+  }, []);
+
+  const showCroppedImage = useCallback(async () => {
+    try {
+      const croppedImage = await getCroppedImg(
+        imageSrc,
+        croppedAreaPixels,
+        rotation
+      );
+      console.log("donee", { croppedImage });
+      setCroppedImageSrc(croppedImage);
 
+      uploadBackground(croppedImage);
+      setTimeout(() => window.location.reload(), 1500);
+      return false;
+    } catch (e) {
+      console.error(e);
+    }
+  }, [imageSrc, croppedAreaPixels, rotation]);
+
+  const onClose = useCallback(() => {
+    setCroppedImage(null);
+  }, []);
+
+  const onFileChange = async (e) => {
+    if (e.target.files && e.target.files.length > 0) {
+      const file = e.target.files[0];
+      let imageDataUrl = await readFile(file);
+
+      setImageSrc(imageDataUrl);
+    }
+  };
   function verifyJpeg(filename) {
     const fnArr = filename.split(".");
     if (fnArr[fnArr.length - 1] == "jpeg" || fnArr[fnArr.length - 1] == "jpg")
       return true;
     return false;
   }
 
-  function uploadBackground(e) {
-    let file = e.target.files[0];
-
-    if (!verifyJpeg(file.name)) return;
+  function uploadBackground(croppedImage) {
+    // let file = e.target.files[0];
+    //  if (!verifyJpeg(file.name)) return;
     const storage = getStorage();
-    const storageRef = ref(storage, "/images/" + file.name);
+    const storageRef = ref(storage, "/images/" + uid());
 
     // 'file' comes from the Blob or File API
-    uploadBytes(storageRef, file).then((snapshot) => {
+    uploadString(storageRef, croppedImage, "data_url").then((snapshot) => {
```

Comment:
Are we verifying if it's a jpeg file again? Seems like the code for that is commented out.